markdown_it/tree.py

```diff
@@ -35,17 +49,17 @@ def __init__(self) -> None:
         self.token: Optional[Token] = None
 
         # Only containers have nester tokens
-        self.nester_tokens: Optional[SyntaxTreeNode._NesterTokens] = None
+        self.nester_tokens: Optional[_NesterTokens] = None
 
         # Root node does not have self.parent
-        self.parent: Optional["SyntaxTreeNode"] = None
+        self.parent = None
 
         # Empty list unless a non-empty container, or unnested token that has
         # children (i.e. inline or img)
-        self.children: List["SyntaxTreeNode"] = []
+        self.children = []
```

Comment:
should this not be `self._children`?

Reply:
There really is no difference, but I think using the public API is less brittle in case someone ever changes the `@property` setter

Reply:
I think all internal variables should be set in `__init__`. TBH I'm suprised flake8 does not fail this, pylint definitely does

Reply:
The internal vars are set in init, but via the @property, flake8 understands this.

Reply:
Reluctantly made this change :smile: